staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -963,3 +965,72 @@ func SelectorFromValidatedSet(ls Set) Selector {
 func ParseToRequirements(selector string, opts ...field.PathOption) ([]Requirement, error) {
 	return parse(selector, field.ToPath(opts...))
 }
+
+// SetSelector wraps a Set, allowing it to implement the Selector interface. Unlike
+// Set.AsSelectorPreValidated (which copies the input Set), this type simply wraps the underlying
+// Set. As a result, it is substantially more efficient, but requires the caller to not mutate the
+// Set.
+// None of the Selector methods mutate the underlying Set, but Add() and Requirements() convert to the
+// less optimized version.
+type SetSelector Set
```

Comment:
I'm wondering if this should be named in a way that indicates the Set is assumed/required to be prevalidated, like `ValidatedSetSelector`

Also, consider copying these bits from `SelectorFromValidatedSet` doc:
```
// A nil and empty Sets are considered equivalent to Everything().
// It assumes that Set is already validated and doesn't do any validation.
```

Reply:
ValidatedSetSelector SGTM, I updated it and the docs